src/appkit.cs

```diff
@@ -22888,5 +23434,68 @@ public interface NSWorkspaceAccessibilityExtensions {
 
 		[Export ("accessibilityDisplayShouldReduceTransparency")]
 		bool AccessibilityDisplayShouldReduceTransparency { get; }
+
+		[Mac (10, 12)]
+		[Export ("accessibilityDisplayShouldInvertColors")]
+		bool AccessibilityDisplayShouldInvertColors { get; }
+
+		[Mac (10, 12)]
+		[Export ("accessibilityDisplayShouldReduceMotion")]
+		bool AccessibilityDisplayShouldReduceMotion { get; }
+	}
+
+	[Mac (10,12)]
+	[BaseType (typeof(NSObject))]
+	interface NSFilePromiseProvider : NSPasteboardWriting
+	{
+		[Export ("fileType")]
+		string FileType { get; set; }
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		NSFilePromiseProviderDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+
+		[NullAllowed, Export ("userInfo", ArgumentSemantic.Strong)]
+		NSObject UserInfo { get; set; }
+
+		[Export ("initWithFileType:delegate:")]
+		IntPtr Constructor (string fileType, NSFilePromiseProviderDelegate @delegate);
+	}
+
+	[Mac (10,12)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
+	interface NSFilePromiseProviderDelegate
+	{
+		[Abstract]
+		[Export ("filePromiseProvider:fileNameForDestination:")]
+		string FilePromiseProvider (NSFilePromiseProvider filePromiseProvider, NSUrl destinationURL);
+
+		[Export ("filePromiseProvider:writePromiseToURL:completionHandler:")]
+		void FilePromiseProvider (NSFilePromiseProvider filePromiseProvider, NSUrl url, Action<NSError> completionHandler);
```

Comment:
same, use `writePromiseToURL:completionHandler:` for something like `WritePromise`